File: app/src/main/java/com/snaptiongame/snaptionapp/data/models/OAuthRequest.java
package com.snaptiongame.snaptionapp.data.models;

/**
 * @author Tyler Wong
 */

public class OAuthRequest {
    public String token;
    public String deviceToken;
    public String deviceType;
    public String linkToken;

    public static final String TOKEN = "token";
    public static final String DEVICE_TOKEN = "device_token";
    public static final String DEVICE_TYPE = "device_type";
    public static final String ANDROID = "Android";
    public static final String LINK_TOKEN = "linkToken";

    public OAuthRequest(String token, String deviceToken, String linkToken) {
        this.token = token;
        this.deviceToken = deviceToken;
        this.deviceType = ANDROID;
        this.linkToken = linkToken;
    }
}

File: app/src/main/java/com/snaptiongame/app/data/models/OAuthRequest.kt
package com.snaptiongame.app.data.models

/**
 * @author Tyler Wong
 */

data class OAuthRequest(var token: String, var deviceToken: String, var linkToken: String) {
    var deviceType: String = ANDROID

    companion object {
        const val TOKEN = "token"
        const val DEVICE_TOKEN = "device_token"
        const val DEVICE_TYPE = "device_type"
        const val ANDROID = "Android"
        const val LINK_TOKEN = "linkToken"
    }
}
